File: practica_2/main.java
*/  
    Practica 2
    Nombre: Alex Gabriel Nina Aliaga
    
    Las clases Linea y Circulo fueron aumentadas al la practica anterior 
*/

/////////////////////////  CLASE PUNTO  /////////////////////////////////////////
class Punto {
    private double x;
    private double y;
    
    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }
    
    public double getX() {
        return x;
    }
    
    public double getY() {
        return y;
    }
    
    public void  cordCartesianas() {
        System.out.println("Las cordenadas cartesianas del punto son: (" + x + ", " + y +")" );
    }
    
    public void  cordPolares() {
        double r = Math.sqrt(x*x + y*y);
        double C = Math.atan(y/x);
        System.out.println("Las cordenadas Polares del punto son : (" + r + ", " + C + ")");
    }
    
    public String imprimirPunto() {
        return "(" + x + ", " + y + ")";
    }
    
    @Override
    public String toString() {
        return "Punto{x = " + x + ",y = " + y +"}";
    }
}

//////////////////////  CLASE LINEA //////////////////////////////////////////

class Linea {
    private Punto p1;
    private Punto p2;
    
    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }
    
    @Override
    public String toString() {
        return "Linea{punto1 = " + p1.imprimirPunto() + " punto2 = " + p2.imprimirPunto() + "}";
    }
    
    public void dibujarLinea() {
        double ekis = p2.getX() - p1.getX();
        double ies = p2.getY() - p1.getY();
        
        double pendiente = ies / ekis;
        double distancia = Math.sqrt((ekis*ekis) + (ies*ies));
        
        System.out.println("La linea tiene una pendiente " + pendiente + " la distancia de punto a punto es " + distancia);
    }
    
}

//////////////////////////// CLASE CIRCULO //////////////////////////////////////////////

class Circulo {
    private Punto centro;
    private float radio;
    
    public Circulo(Punto centro, float radio) {
        this.centro = centro;
        this.radio = radio;
    }
    
    @Override
    public String toString() {
        return "Circulo{centro = " + centro + " radio = " + radio;
    }
    
    public void dibujarCirculo() {
        double area = 3.14 * Math.pow(radio, 2);
        System.out.println("El area del circulo " + area + " con centro en el " + centro);
    }
}

///////////////////////// CLASE PRINCIPAL (IMPORTANTE)//////////////////////////

public class Main
{
	public static void main(String[] args) {
		Punto pp1 = new Punto(3, 4);
		Punto pp2 = new Punto(6, 5);
		
		System.out.println(pp1);
		pp1.cordCartesianas();
		pp1.cordPolares();
		
		System.out.println("--------------------------------------------");
		
		Linea l1 = new Linea(pp1, pp2);
		System.out.println(l1);
		l1.dibujarLinea();
		
		System.out.println("--------------------------------------------");
		
		Circulo c1 = new Circulo(pp1, 3);
		System.out.println(c1);
		c1.dibujarCirculo();
		
		
	}
}

File: practica_2/main.py
import math
import matplotlib.pyplot as plt
import numpy as np

# Clase Punto
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def coord_cartesianas(self):
        print(f"Las coordenadas cartesianas del punto son: ({self.x}, {self.y})")

    def coord_polares(self):
        r = math.sqrt(self.x * self.x + self.y * self.y)
        C = math.atan2(self.y, self.x)
        print(f"Las coordenadas polares del punto son: ({r}, {C})")

    def imprimir_punto(self):
        return f"({self.x}, {self.y})"

    def __str__(self):
        return f"Punto{{x = {self.x}, y = {self.y}}}"

# Clase Linea
class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Linea{{punto1 = {self.p1.imprimir_punto()} punto2 = {self.p2.imprimir_punto()}}}"

    def dibujar_linea(self, ax):
        x_values = [self.p1.get_x(), self.p2.get_x()]
        y_values = [self.p1.get_y(), self.p2.get_y()]
        ax.plot(x_values, y_values, 'b-', label="Línea")  # Línea azul
        ax.scatter(x_values, y_values, color='red', zorder=3)  # Puntos en rojo

        ekis = self.p2.get_x() - self.p1.get_x()
        ies = self.p2.get_y() - self.p1.get_y()
        pendiente = ies / ekis if ekis != 0 else None
        distancia = math.sqrt((ekis * ekis) + (ies * ies))

        print(f"La línea tiene una pendiente {pendiente} y la distancia de punto a punto es {distancia}")

# Clase Circulo
class Circulo:
    def __init__(self, centro, radio):
        self.centro = centro
        self.radio = radio

    def __str__(self):
        return f"Circulo{{centro = {self.centro} radio = {self.radio}}}"

    def dibujar_circulo(self, ax):
        area = math.pi * math.pow(self.radio, 2)
        print(f"El área del círculo es {area} con centro en {self.centro}")

        # Dibujar círculo en la gráfica
        circle = plt.Circle((self.centro.get_x(), self.centro.get_y()), self.radio, color='green', fill=False)
        ax.add_patch(circle)
        ax.scatter(self.centro.get_x(), self.centro.get_y(), color='red', zorder=3)  # Centro en rojo

# Función principal
def main():
    pp1 = Punto(3, 4)
    pp2 = Punto(6, 5)

    print(pp1)
    pp1.coord_cartesianas()
    pp1.coord_polares()

    print("--------------------------------------------")

    l1 = Linea(pp1, pp2)
    print(l1)

    print("--------------------------------------------")

    c1 = Circulo(pp1, 3)
    print(c1)

    # Configurar la gráfica
    fig, ax = plt.subplots()
    ax.set_xlim(0, 10)
    ax.set_ylim(0, 10)
    ax.set_aspect('equal', adjustable='datalim')
    ax.grid(True)

    # Dibujar elementos
    l1.dibujar_linea(ax)
    c1.dibujar_circulo(ax)

    plt.legend()
    plt.show()

# Ejecutar la función principal
main()
